core/src/main/java/nl/nn/adapterframework/align/XmlTypeToJsonSchemaConverter.java

```diff
@@ -89,10 +110,30 @@ public JsonStructure createJsonSchema(String elementName, String namespace) {
 				}
 			return null;
 		}
-		return createJsonSchema(elementName, elementDecl);
+
+		JsonObjectBuilder builder = Json.createObjectBuilder();
+		builder.add("$schema", JSON_SCHEMA);
+		if(elementDecl.getNamespace() != null){
+			builder.add("$id", elementDecl.getNamespace());
+		}
+		if(schemaLocation != null){
+			builder.add("description", "Auto-generated by Frank!Framework based on " + schemaLocation);
+		}
+		if (skipRootElement) {
+			builder.add("$ref", definitionsPath+elementName);
+		} else {
+			addType(builder, "object").add("additionalProperties", false);
+			builder.add("properties", Json.createObjectBuilder().add(elementName, Json.createObjectBuilder().add("$ref", definitionsPath+elementName).build()));
+		}
+		JsonObject definitionsBuilderResult = getDefinitions();
+		if(!definitionsBuilderResult.isEmpty()){
+			builder.add("definitions", definitionsBuilderResult);
+		}
+		return builder.build();
```

Comment:
Why has this been moved? it might be better to leave this in it's own method.

Reply:
moved by cleaning things up. The method is only called from this method and not too complex.

Reply:
I understand it's not too complex now, but if we expand on this code (within this method) it will.